src/Serilog.Sinks.Seq/Sinks/Seq/HttpLogShipper.cs

```diff
@@ -280,6 +282,58 @@ void OnTick()
             }
         }
 
+        private const string _invalidPayloadFilePrefix = "invalid-";
+        private async Task FailResponse(HttpResponseMessage result, string payload)
+        {
+            var invalidPayloadFilename = $"{_invalidPayloadFilePrefix}{result.StatusCode}-{Guid.NewGuid():n}.json";
+            var invalidPayloadFile = Path.Combine(_logFolder, invalidPayloadFilename);
+            var resultContent = await result.Content.ReadAsStringAsync();
+            SelfLog.WriteLine("HTTP shipping failed with {0}: {1}; dumping payload to {2}", result.StatusCode, resultContent, invalidPayloadFile);
+            var bytesToWrite = Encoding.UTF8.GetBytes(payload);
+            if (_errorFileSizeLimitBytes.HasValue)
+            {
+                CleanupErrorFiles(_errorFileSizeLimitBytes.Value - bytesToWrite.Length, _logFolder);
+            }
+            IOFile.WriteAllBytes(invalidPayloadFile, bytesToWrite);
+        }
+
+        private static void CleanupErrorFiles(long maxNumberOfBytesToRetain, string logFolder)
+        {
+            try
+            {
+                var candiateFiles = Directory.EnumerateFiles(logFolder, $"{_invalidPayloadFilePrefix}*.json");
+                DeleteOldFiles(maxNumberOfBytesToRetain, candiateFiles);
+            }
+            catch (Exception ex)
+            {
+                SelfLog.WriteLine("Exception thrown while trying to CleanupErrorFiles: {0}", ex);
+            }
+        }
+
+        private static void DeleteOldFiles(long maxNumberOfBytesToRetain, IEnumerable<string> files)
+        {
+            long runningTotal = 0L;
+
+            var invalidPayloadFilesToDelete = from candiateFile in files
+                                              let candiateFileInfo = new FileInfo(candiateFile)
+                                              orderby candiateFileInfo.LastAccessTimeUtc descending
+                                              let cumulative = runningTotal += candiateFileInfo.Length
```

Comment:
Since `runningTotal` is mutated imperatively here, enumeration of `invalidPayloadFilesToDelete` more than once will result in an invalid total. Not an issue in the code as written, but may be a future footgun.

It's a nice succinct bit of code, but is there a way to express the same algorithm without the gotcha? I can't come up with a quick transformation that would do it, but keeping the first three lines here (from/let/orderby) declarative and converting the rest to a `while` loop should do the trick.